src/services/ipinfo-client.ts

```diff
@@ -0,0 +1,21 @@
+import axios from 'axios';
+
+export interface IpInfoResult {
+  data: { country: string };
+}
+
+export interface IpinfoClient {
+  getCountryCode(): Promise<IpInfoResult>;
+}
+
+export class HttpIpinfoClient implements IpinfoClient {
+  public async getCountryCode(): Promise<IpInfoResult> {
+    try {
+      const response = await axios.get('http://ipinfo.io/json');
+      const countryCode = response.data && response.data.country ? response.data.country : 'AR';
+      return { data: { country: countryCode } };
+    } catch (error) {
+      return { data: { country: 'AR' } };
```

Comment:
Oh, you are already dealing with the fallback here, maybe you can simplify the result format.